tool/tctl/common/plugins_command.go

```diff
@@ -96,11 +166,146 @@ func (p *PluginsCommand) Cleanup(ctx context.Context, clusterAPI *auth.Client) e
 	return nil
 }
 
+type samlConnectorsClient interface {
+	GetSAMLConnector(ctx context.Context, id string, withSecrets bool) (types.SAMLConnector, error)
+}
+
+type pluginsClient interface {
+	CreatePlugin(ctx context.Context, in *pluginsv1.CreatePluginRequest, opts ...grpc.CallOption) (*emptypb.Empty, error)
+}
+
+type installPluginArgs struct {
+	samlConnectors samlConnectorsClient
+	plugins        pluginsClient
+}
+
+func (p *PluginsCommand) InstallOkta(ctx context.Context, args installPluginArgs) error {
+	oktaSettings := p.install.okta
+
+	if oktaSettings.accessListSync {
+		if len(oktaSettings.defaultOwners) == 0 {
+			return trace.BadParameter("AccessList sync requires at least one default owner to be set")
+		}
+	}
+
+	if oktaSettings.scimToken != "" {
+		if len(oktaSettings.defaultOwners) == 0 {
+			return trace.BadParameter("SCIM support requires at least one default owner to be set")
+		}
+
+		if oktaSettings.appID == "" {
+			return trace.BadParameter("SCIM support requires app-id to be set")
+		}
+
+		if oktaSettings.samlConnector == "" {
+			return trace.BadParameter("SCIM support requires saml-connector to be set")
+		}
+	}
+
+	if oktaSettings.samlConnector != "" {
+		if err := validateSAMLConnector(ctx, args.samlConnectors, oktaSettings.samlConnector); err != nil {
+			fmt.Printf("Failed validating SAML connector: %v", err)
+			return trace.Wrap(err)
+		}
+	}
+
+	creds := []*types.PluginStaticCredentialsV1{
+		{
+			ResourceHeader: types.ResourceHeader{
+				Metadata: types.Metadata{
+					Name: p.install.name,
+					Labels: map[string]string{
+						types.OktaCredPurposeLabel: types.OktaCredPurposeAuth,
+					},
+				},
+			},
+			Spec: &types.PluginStaticCredentialsSpecV1{
+				Credentials: &types.PluginStaticCredentialsSpecV1_APIToken{
+					APIToken: oktaSettings.apiToken,
```

Comment:
How this will work for zoom integration ? 

Does creating a `PluginStaticCredentialsSpecV1_APIToken` without `APIToken` is even allowed ?